dev/TabView/TabView.cpp

```diff
@@ -233,14 +233,42 @@ void TabView::OnListViewLoaded(const winrt::IInspectable&, const winrt::RoutedEv
         // Now that ListView exists, we can start using its Items collection.
         auto items = TabItems();
         auto numItemsToCopy = static_cast<int>(items.Size());
-        if (auto lvItems = listView.Items())
+        bool itemsSourceDoesNotExists = listView.ItemsSource() == nullptr;
+
+        if (itemsSourceDoesNotExists)
+        {
+            if (auto lvItems = listView.Items())
+            {
+                // copy the list, because clearing lvItems may also clear items
+                winrt::IVector<winrt::IInspectable> item_list{ winrt::single_threaded_vector<winrt::IInspectable>() };
+
+                for (auto itm : items)
+                {
+                    item_list.Append(itm);
+                }
+
+                lvItems.Clear();
+
+                for (int i = 0; i < numItemsToCopy; i++)
+                {
+                    // App put items in our Items collection; copy them over to ListView.Items
+                    auto itm = item_list.GetAt(i);
+                    if (itm != nullptr)
+                        lvItems.Append(itm);
+                }
+                TabItems(lvItems);
+            }
+        }
+
+        else if (auto itemsSource = listView.ItemsSource().try_as<winrt::IIterable<winrt::IInspectable>>())
         {
-            for (int i = 0; i < numItemsToCopy; i++)
+            winrt::IVector<winrt::IInspectable> item_list{ winrt::single_threaded_vector<winrt::IInspectable>() };
+
+            for (auto itm : itemsSource)
             {
-                // App put items in our Items collection; copy them over to ListView.Items
-                lvItems.Append(items.GetAt(i));
+                item_list.Append(itm);
             }
-            TabItems(lvItems);
+            TabItems(item_list);
```

Comment:
I'm not quite understanding this "else if" block. If there is an ItemsSource, why are we copying those items into TabItems? TabItems is no longer the same as ListView.Items?

Reply:
You are right, actually the else part is unnecessary. Thanks.

Reply:
https://github.com/microsoft/microsoft-ui-xaml/pull/1482#discussion_r338765792